core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -117,49 +118,54 @@ public BlockInfo getInfo(long blockId) {
    * Gets a stream to read the data of a block. The stream is backed by Alluxio storage.
    *
    * @param blockId the block to read from
+   * @param openUfsBlockOptions the options to open UFS block, set to null if the block is not in
+   *        UFS
    * @param options the options
    * @return an {@link InputStream} which can be used to read the data in a streaming fashion
    */
-  public BlockInStream getInStream(long blockId, InStreamOptions options) {
+  public BlockInStream getInStream(long blockId, Protocol.OpenUfsBlockOptions openUfsBlockOptions,
+      InStreamOptions options) {
     BlockInfo blockInfo;
     try (CloseableResource<BlockMasterClient> masterClientResource =
         mContext.acquireBlockMasterClientResource()) {
       blockInfo = masterClientResource.get().getBlockInfo(blockId);
     }
 
-    if (blockInfo.getLocations().isEmpty()) {
+    if (blockInfo.getLocations().isEmpty() && openUfsBlockOptions == null) {
       throw new NotFoundException("Block " + blockId + " is not available in Alluxio");
```

Comment:
should we update this message? it is not in Alluxio and not known how to open in UFS?